### PR TITLE
Fix mermaid event based

### LINKED ISSUE #35: Fixing the mermaid code for linear and event based
For the event based SMF -> get the code from tables, Display in Chainlit
For the linear -> Have the GSM be connected

### PR DESCRIPTION
**Pull request by Kalamar136**
Closes #35

### CODE REVIEW

File: backend/event_driven_smf/event_driven_smf.py [hunk 1]
```diff
@@ -103,7 +107,19 @@
 qa_agent = QAAgent(llm=llm, belief=belief, num_runs=100, policy=policy)
 
 def run_event_driven_smf():
-    qa_agent.run()
+    with open(f'{os.path.dirname(__file__)}\\..\\resources\\event_driven_log\\output_event_driven{time.strftime("%m_%d_%H_%M_%S")}.txt', 'w') as f:
```
Comment (AlecTufenkjian) [inline backend/event_driven_smf/event_driven_smf.py:110]: image

This line is strange, can we make it simpler, and more generalized? 

I can't run it, could you help me how I'm supposed to run it?

Comment (AlecTufenkjian) [reply backend/event_driven_smf/event_driven_smf.py:110]: image

Comment (Kalamar136) [reply backend/event_driven_smf/event_driven_smf.py:110]: Hmm it's weird it works for me
Basically the relative path is relying on the repo structure
I had to add the double '\' to make it work

Comment (Kalamar136) [reply backend/event_driven_smf/event_driven_smf.py:110]: Ok actually we fixed it together by committing the empty directories with the .keep files.
Please let me know if it works for you now

**Pull request merged**